solr/core/src/java/org/apache/solr/cloud/ZkController.java

```diff
@@ -2627,8 +2626,7 @@ private static void ensureRegisteredSearcher(SolrCore core) throws InterruptedEx
         }
         registeredSearcher.decref();
       } else  {
-        @SuppressWarnings({"rawtypes"})
-        Future[] waitSearcher = new Future[1];
+        AtomicReference<Future<Void>> waitSearcher = new AtomicReference<>();
```

Comment:
We can do this, or we can do Array.newInstance and cast like I've done in other places, no strong opinions although I do lean towards the Reference solution because it feels like the intent is clearer to me.